### PR TITLE
Added service for ensuring system clock is plausibly set

### PR DESCRIPTION
**Pull request by moduspwnens**
### Background

I'm using two Raspberry Pis at home running k3os as single node clusters and ran into the following issue:

 * The lack of a hardware clock means the Raspberry Pi relies entirely on an NTP server at boot to set the system clock
 * k3s is [hard-coded to fail to start](https://github.com/rancher/k3s/blob/66a8c2ad7f0f9b87377aeb06201d1dab3007fdde/pkg/cli/cmds/log.go#L82) if the system time is set before 1980
 * There's no mechanism currently in k3os to wait for the clock to be validly set before starting k3s
 * After failing to start k3s some amount of times, it will no longer be retried

This means if, for whatever reason, the Pi is unable to update its system time quickly enough via NTP, it can go into a state that requires administrator intervention to recover.

In my case, these Pis are serving as the DNS servers for my network, so it's important to me that they are sufficiently robust that they will begin running Kubernetes soon after (e.g.) a network-wide power outage is restored. They're configured to sync time with a public NTP server, but my cable modem can sometimes take a few minutes to initialize Internet connectivity. The net result is that their clocks don't get synced for a few minutes when starting up, but by then k3s has failed to start enough times that it's no longer being retried.

### TL;DR

I created this pull request to cause k3os to wait until the system time is set to 1980 or later before starting k3s. This based on the idea that there's no plausible circumstance under which one would want k3s to start without that happening, since it will definitively fail due to the hard-coded check.

### Details

This should have no noticeable effect on any system with a functioning / set hardware clock.

I did set a configurable timeout for it in `/etc/conf.d/clock-sane`, although it is set to 0 (wait forever) by default.

The only other change here is to move sshd to the `boot` runlevel (instead of `default`). This was done to ensure sshd isn't delayed in coming up as a result of waiting for the system time to be set.

### CODE REVIEW

**Comment by moduspwnens** [comment]
Thanks for your time in reviewing.

I just adapted this from the net-online service script already in this project. I don't know what other distros are using--they might not be doing anything. I suspect it's a little of a unique case with k3s being hardcoded to fail to start with the system time before 1980.

I'll review and try to resolve the issue with sshd start ordering without changing its runlevel. Thank you
